plugins/vpc-bridge/config/k8s/pipe.go

```diff
@@ -0,0 +1,197 @@
+	p.setResultChan(resultChan) // Set results chan on Pipe object
+
+	// Accept connections and read pipe in go routine as these operations are blocking.
+	go p.acceptConnections(listener, resultChan)
+
+	return nil
+}
+
+// GetPath get name/path for named pipe.
+func (p *Pipe) GetPath() string {
+	return p.Path
+}
+
+// GetOutput Get output received from named pipe using results chan.
+func (p *Pipe) GetOutput(ctx context.Context) (string, error) {
+
+	// Get result chan from Pipe object.
+	resultChan := p.getResultChan()
+
+	// Get output from named pipe using result chan.
+	var result PipeReadResult
+	select {
+	case result = <-resultChan:
+	case <-ctx.Done():
+		return "", fmt.Errorf("timeout getting output from named pipe result chan")
+	}
+	return result.stdout, result.stderr
+}
+
+// Close closes pipe listener.
+func (p *Pipe) Close() error {
+	err := p.getListener().Close()
+	if err != nil {
+		return fmt.Errorf("error closing pipe listener: %w", err)
+	}
+	return nil
+}
+
+// setListener set named pipe listener to Pipe object.
+func (p *Pipe) setListener(listener net.Listener) {
+	p.listener = listener
+}
+
+// getListener get named pipe listener from Pipe object.
+func (p *Pipe) getListener() net.Listener {
+	return p.listener
+}
+
+// setResultChan set results chan to Pipe object.
+func (p *Pipe) setResultChan(outputChan chan PipeReadResult) {
+	p.resultChan = outputChan
+}
+
+// getResultChan get results chan from Pipe object.
+func (p *Pipe) getResultChan() chan PipeReadResult {
+	return p.resultChan
+}
+
+// listen creates named pipe. creates listener on named pipe.
+// Returns listener for named pipe.
+func (p *Pipe) listen() (net.Listener, error) {
+	pipeName := p.GetPath()
+	// Create pipe and get listener.
+	// listener is closed in main routine.
+	pipeListener, err := winio.ListenPipe(pipeName, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating named pipe: %s %w", pipeName, err)
+	}
+	log.Debugf("named piped: %s created", p.GetPath())
+	return pipeListener, nil
+}
+
+// acceptConnections start accepting connections and read message from named pipe.
+// This runs in go routine as Accept and Read from conn are blocking operations.
+// Result is sent over result channel. Added Timeout for read operation on conn.
+func (p *Pipe) acceptConnections(pipeListener net.Listener, result chan PipeReadResult) {
+	// Get named pipe path.
+	pipNamePath := p.GetPath()
+
+	// Accept connection on named pipe. This operation is blocking until client dials into pipe.
+	conn, err := pipeListener.Accept()
+	if err != nil {
+		log.Errorf("error accepting connection on pipe listener: %v %w", pipNamePath, err)
+		// Send error over channel.
+		result <- PipeReadResult{stderr: fmt.Errorf("error accepting connection on pipe listener: %v %w", pipNamePath, err)}
+		return
+	}
+	defer conn.Close() // Defer closing connection.
+
+	// Setup read timeout on conn object.
+	err = conn.SetReadDeadline(time.Now().Add(PipeConnectionTimeout))
```

Comment:
The context in Start() has the appropriate timeout. You should use it here and not create a different arbitrary timeout

Reply:
done, `conn` now uses deadline from context.